Ch04/drill4.cpp

```diff
@@ -0,0 +1,52 @@
+#include "../std_lib_facilities.h"
+
+double sum(const vector<double>& v) {
+	double sumofnum = 0.0;
+	for (double i:v) {
+		sumofnum += i;
+	};
+	return sumofnum;
+}
+
+int main () {
+	string sint1;
+	double int1;
+	vector<double> meters;
+
+	string unit;
+	constexpr double cmtom = 1.0/100;
+	constexpr double intom = 2.54/100;
+	constexpr double fttom = 12*2.54/100;
+	
+	while (true) {
+		cout << "Type a measurement and a unit (m, cm, in, ft). ";
+		cin >> sint1 >> unit;
+		if (sint1 == "|") {
+			break;
+		}
+		int1 = stod(sint1);
+		if (unit == "cm") {
```

Comment:
I would create 2 functions out of this:

    bool supportedUnit(string unit) {
        ..
    }

    double conversionFactor(string unit) {
        ...
    }

Then in main(), I can write code like this:

    if supportedUnit(unit) {
        meters.push_back(int1 *= conversionFactor(unit))
    } else {
        cout << "That is not a valid measurement.\n";
        continue;
    }